Core/GDCore/Project/Project.cpp

```diff
@@ -97,24 +95,62 @@ std::unique_ptr<gd::Object> Project::CreateObject(
                      " has an unknown default behavior: " + behaviorType);
       return;
     }
-    auto* behavior = object->AddNewBehavior(
-        project, behaviorType, behaviorMetadata.GetDefaultName());
-    behavior->SetDefaultBehavior(true);
+
+    const gd::String& behaviorName = behaviorMetadata.GetDefaultName();
+
+    // Check if we can keep a behavior that would have been already set up on the object.
+    if (object.HasBehaviorNamed(behaviorName)) {
+      const auto& behavior = object.GetBehavior(behaviorName);
+
+      if (!behavior.IsDefaultBehavior() || behavior.GetTypeName() != behaviorType) {
+        // Behavior type has changed, remove it so it is re-created.
+        object.RemoveBehavior(behaviorName);
+      }
+    }
+
+    if (!object.HasBehaviorNamed(behaviorName)) {
+      auto* behavior = object.AddNewBehavior(
+          project, behaviorType, behaviorName);
+      behavior->SetDefaultBehavior(true);
+    }
   };
 
   auto &objectMetadata =
       gd::MetadataProvider::GetObjectMetadata(platform, objectType);
   if (!MetadataProvider::IsBadObjectMetadata(objectMetadata)) {
-    for (auto &behaviorType : objectMetadata.GetDefaultBehaviors()) {
+    // Add all default behaviors.
+    const auto& defaultBehaviorTypes = objectMetadata.GetDefaultBehaviors();
+    for (auto &behaviorType : defaultBehaviorTypes) {
       addDefaultBehavior(behaviorType);
     }
+
+    // Ensure there are no default behaviors that would not be required left on the object.
+    for (const auto& behaviorName : object.GetAllBehaviorNames()) {

```

Comment:
It's probably not a big issue, but instance stability won't be ensured for capabilities required for an object without any type.
Maybe we can add a specific case for this.

Reply:
Capabilities from parameters are indeed not marked as default behaviors.